src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -15,10 +22,150 @@
  * GridFieldFilterHeader alters the {@link GridField} with some filtering
  * fields in the header of each column.
  *
+ * filter header with customisable filter fields
+ *
+ * REQUIREMENTS:
+ *
+ * table header component needs to be present in the GridField (for example GridFieldSortableHeader)
+ * the last column of the table needs to have a vacant header so the filter widget could be displayed there
+ * for example you can't have the last column with sorting header and filter widget at the same time
+ *
+ * ALIAS FIELDS
+ *
+ * in the case that the column name of the table is different from matching field name via the summary_fields
```

Comment:
In the SilverStripe ORM Column === Field name. I think you may need to clarify this.

A better example would be a summary_field with humanised label.

E.g.

```
private static $summary_fields = [
  'getTitleSummary' => 'Summary Title',
  'City.Name' => 'City Name',
  'Expires.Nice' => 'Expires on',
];
```